pkg/genericapiserver/server/storage_factory.go

```diff
@@ -244,15 +253,23 @@ func (s *DefaultStorageFactory) NewConfig(groupResource schema.GroupResource) (*
 	return &config, nil
 }
 
-// Get all backends for all registered storage destinations.
+// Backends returns all backends for all registered storage destinations.
 // Used for getting all instances for health validations.
-func (s *DefaultStorageFactory) Backends() []string {
-	backends := sets.NewString(s.StorageConfig.ServerList...)
+func (s *DefaultStorageFactory) Backends() []Backend {
+	servers := sets.NewString(s.StorageConfig.ServerList...)
 
 	for _, overrides := range s.Overrides {
-		backends.Insert(overrides.etcdLocation...)
+		servers.Insert(overrides.etcdLocation...)
+	}
+	backends := []Backend{}
+	for server := range servers {
+		backends = append(backends, Backend{
+			Server:   server,
+			KeyFile:  s.StorageConfig.KeyFile,
```

Comment:
Yeah, all the configurations are inherited from `storagebackend.Config`, if they have to use different one, we will need to update that part as well..